File: apps/parser/src/com/eg/model/Beverage.java
package com.eg.model;

import org.apache.commons.csv.CSVRecord;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * Created by Alex Romayev on 12/5/16.
 */
public class Beverage extends Product {
    private String segment;
    private String base;
    private String starchUseLabel;
    private String fatContent;
    private String proteinContent;
    private String features;
    private String productDescription;
    private List<Product> related;

    public Beverage(CSVRecord record) {
        super(record);
        segment = get(record, "Segment");
        base = get(record, "Base");
        starchUseLabel = get(record, "% Starch use level");
        fatContent = get(record, "Fat content");
        proteinContent = get(record, "Protein Content");
        productDescription = get(record, "Product Description");
        features = get(record, "Features");
        readValueProposition(record);
    }

    @Override
    void readValueProposition(CSVRecord record) {
        valueProposition = get(record, "Value Proposition 1");
        String valueProposition2 = get(record, "Value Proposition 2");
        addRelated(valueProposition2);

        String valueProposition3 = get(record, "Value Proposition 3");
        addRelated(valueProposition3);
    }

    private Beverage(Beverage other) {
        super(other);
        segment = other.segment;
        base = other.base;
        starchUseLabel = other.starchUseLabel;
        fatContent = other.fatContent;
        proteinContent = other.proteinContent;
        features = other.features;
        productDescription = other.productDescription;
    }

    private void addRelated (String valueProposition) {
        if (valueProposition != null && valueProposition.length() > 0) {
            if (related == null) {
                related = new ArrayList<Product>();
            }
            Beverage other = new Beverage(this);
            other.valueProposition = valueProposition;
            related.add(other);
        }
    }

    public List<Product> getRelated() {
        return related;
    }

    @Override
    public Map<String, Object> plist() {
        Map<String, Object> map = super.plist();
        map.put("segment", segment);
        map.put("base", base);
        map.put("starchUseLabel", starchUseLabel);
        map.put("fatContent", fatContent);
        map.put("proteinContent", proteinContent);
        map.put("features", features);
        map.put("productDescription", productDescription);
        return map;
    }
}

File: apps/Ingredion/Ingredion/Classes/Model/Beverage.swift
//
//  Beverage.swift
//  Ingredion
//
//  Created by Alex Romayev on 2/10/17.
//  Copyright © 2017 Executive Graphics. All rights reserved.
//

import Foundation

class Beverage : Product {
    @objc let segment: String?
    @objc let base: String?
    let starchUseLabel: String?
    let fatContent: String?
    let proteinContent: String?
    let features: String?
    let productDescription: String?

    override init(_ dictionary: Dictionary<String, Any>) {
        segment = dictionary["segment"] as? String
        base = dictionary["base"] as? String
        starchUseLabel = dictionary["starchUseLabel"] as? String
        fatContent = dictionary["fatContent"] as? String
        proteinContent = dictionary["proteinContent"] as? String
        features = dictionary["features"] as? String
        productDescription = dictionary["productDescription"] as? String
        super.init(dictionary)
        productType = .beverages
    }
}
